### DISCUSSION TITLE
[SOLVED] How to set the value of a different column for a Nested DataGrid SelectList

### DISCUSSION THREAD

**Discussion by gac-noob** [Megabit/Blazorise#1872] [Q&A] [answered]
Hi,

I have a DataGrid for OrderItems that contains 2 SelectList..1 for Product, and 1 for Product Unit of Measures...my goal is that when a user selects a product, the unit of measures will display only the unit of measures for the selected product.

My razor code looks like this:
```
<DataGridColumn Caption="Item" Editable="true" TItem="OrderItem" Field="@nameof(OrderItem.ProductId)">
     <DisplayTemplate>
          @($"{(ProductList.Where(w => w.Id == ((context as OrderItem).ProductId)).FirstOrDefault())?.Name}")
     </DisplayTemplate>
     <EditTemplate>
          <SelectList TItem="ProductListItem"
                    TValue="int?"
                    Data="@ProductList"
                    TextField="@(i => i.Name)"
                    ValueField="@(i => i.Id)"
                    SelectedValue="@((int?)context.CellValue)"
                    SelectedValueChanged="@((v) => OnProductValueChange(context, v))" />
     </EditTemplate>
</DataGridColumn>
<DataGridColumn Caption="Unit of Measure" Editable="true" TItem="OrderItem" Field="@nameof(OrderItem.ProductUnitId)">
     <DisplayTemplate>
          @($"{(ProductUnitList.Where(w => w.Id == ((context as OrderItem).ProductUnitId)).FirstOrDefault())?.Name}")
     </DisplayTemplate>
     <EditTemplate>
          <SelectList TItem="ProductUnitListItem"
                    TValue="int?"
                    Data="@FilteredProductUnitList"
                    TextField="@(i => i.Name)"
                    ValueField="@(i => i.Id)"
                    SelectedValue="@((int?)context.CellValue)"
                    SelectedValueChanged="(v => ((CellEditContext)context).CellValue = v)" />
     </EditTemplate>
</DataGridColumn>
```

And my code block looks like this:

```
private void OnDataGridNewItemDefaultSetter(OrderItem newRecord)
{
     newRecord.ProductId = ProductList.FirstOrDefault().Id; 
     FilterProductUnitList(newRecord.DocProductId);
     newRecord.ProductUnitId = FilteredProductUnitList.FirstOrDefault().Id;
}

private void OnProductValueChange(CellEditContext context, int? selectedProductId)
{
     context.CellValue = selectedProductId;
     FilterProductUnitList(selectedProductId);	

     // This is where the problem is..how do I access the ProductUnitId of the selected row?
}
```
_Note: I did not include the code for_ `FilterProductUnitList()` _to not clutter my question with unnecessary details, but the function works._

My problem is that in `OnDataGridNewItemDefaultSetter()` I have access to the new `OrderItem.ProductUnitId ` but in `OnProductValueChange()` I only have access to the `CellEditContext ` and it contains the `ProductId `only and does not include the `ProductUnitId `

**Is there a way for me to access the row's context so that I can set the values of other columns?**

**Comment by stsrki** [comment, marked as answer]
You also have access to the `Item` in the `EditTemplate` context, eg. `context.Item`. Note however that item is only available in 0.9.3 that is coming soon. In case you want to try it now you can get it from https://github.com/stsrki/Blazorise#try-preview

**Reply by gac-noob** [reply]
Got it sir Mladen, so for versions below 0.9.3, `Item `in the `EditTemplate `context is not yet available...I'll definitely check out your 0.9.3 preview and will also wait for its official release!!! :)

**Reply by stsrki** [reply]
Yep, it is the new API I introduced recently. Just remember, Item is used only for **read**, do not **write** to it. Use `context.CellValue` as always.

**Reply by gac-noob** [reply]
ohhhh okay...going back to my original problem sir Mladen, let's say I have a list of Products and Unit Of Measures
```
     Products               Unit of Measures
     ------------------     -----------------
     [1] Oranges            [1] Pieces
                            [2] Dozens
     [2] Cooking Oil        [3] Liter
                            [4] Gallon
```

When user selects **[1] Oranges**, Unit of Measures will be filtered to include and display **[1] Peices** and **[2] Dozens** only..

As of this point, the user has **not** yet selected a Unit of Measure, but the `SelectList` already displays **[1] Pieces**  in the screen because it is the 1st item on the list...I need to store the selected unit of measure id to `UnitId`, but I am still on the `OnProductChange` event..

And if the use selects again another product, say Cooking Oil, the Unit of Measures will be filtered again to include and display **[3] Liter** and **[3] Gallon** only..and the `SelectList` will display **[3] Liter** since it is the 1st item on the list..

Everytime a different product is selected, the unit of measure list will be changed, and I will need to store the 1st id of the unit of measure `SelectList` to `UnitId`, but the problem is I am still on the `OnProductChange` event..

I am able to correctly filter the lists, but I can't write the UnitId... :( Any suggestions sir?

**Reply by stsrki** [reply]
To tell you the truth nothing comes to mind. You might try writing to `Item.UnitId` in this case(just this once). I have not too much time to test, it might or might not work.

**Reply by gac-noob** [reply]
hehehe it's okay sir Mladen, thank you very much for the time and effort to respond :) it is greatly appreciated!
